### ISSUE TITLE
[Feature Request] Scroll tree-view entry into view *without autoReveal*

### ISSUE THREAD

**Issue by Redsandro** [atom/tree-view#1246]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

The feature to automatically scroll-into-view the tree-view entry on tab-click is desirable. However, it doesn't happen when _autoReveal_ is disabled.

We like to manually toggle/reveal the tree-view, but (optionally) have the currently selected tab entry centered in the tree-view _once we manually toggle the tree-view_.

### Steps to Reproduce

1. disable autoReveal
2. switch source file tabs
3. manually reveal tree-view

**Expected behavior:** Source file in switched tab is now the visibly selected and centered entry in tree view

**Actual behavior:** Tree-view entry is selected (highlighted) but not scrolled into view.

Perhaps we can have a boolean option `scrollIntoView` for this, and it can be adhered to independent of the value of `autoReveal`, so that the behavior with or without autoReveal is the same. This way autoReveal would only indicate wether or not the tree-view should show itself when hidden or not.

### Versions

```
Atom    : 1.24.1
Electron: 1.6.16
Chrome  : 56.0.2924.87
Node    : 7.4.0

apm  1.18.12
npm  3.10.10
node 6.9.5 x64
atom 1.24.1
python 2.7.12
git 2.7.4

Ubuntu 16.04.3 LTS x64
```

**Comment by Ben3eeE** [comment]
Thanks for the report!

I think the expected behavior you describe is the expected behavior and this is a bug report :sweat_smile: 

I am unable to reproduce the behavior you describe when running the latest master version of tree-view. The manually revealed file is scrolled into view independent off the `Auto Reveal` and `Focus on Reveal` settings even if the tree-view is not visible when revealing.

These options changed a bit in the tree-view version that is on the latest beta. Can you try in the latest beta to see if you can reproduce this behavior? See https://flight-manual.atom.io/getting-started/sections/installing-atom/#debian-and-ubuntu-debapt for how to install the Atom beta from official sources on ubuntu :slightly_smiling_face:

**Comment by Redsandro** [comment]
Hey @Ben3eeE, thanks for the quick reply!

Can I install atom-beta (completely) separate from atom?

**Comment by Ben3eeE** [comment]
They share settings and packages through the `~/.atom` folder so it's not completely separate. But it should be possible to install both of them. You use `atom-beta` to run the beta and `atom` to run stable.

**Comment by Redsandro** [comment]
@Ben3eeE I have checked out `atom-beta` on a clean Ubuntu 18.04 installation and I found that the tree-view still doesn't work as expected (making this a bug), or at least not as I expected (making this a feature request).

I assume you are seeing the same behavior, so I would guess that this is not a bug, but rather a difference in our intuitive expectations and/or preferences.

### Steps to reproduce:

_(insert giggle about header)_

* Disable `autoReveal`
* Open a source directory with enough files for a scrollbar to appear in `tree-view`.
* Open a file/editor/pane for a file somewhere in the top of the list and one somewhere in the bottom of the list
* Switch between tabs
* :heavy_check_mark: Observe the selected entry in tree-view changes.
* :no_entry_sign: Observe the selected entry does NOT scroll into view.

### Expected behavior:

Newly selected tree-view entry should always be scrolled into view (in case of bug), or if some people prefer that this doesn't happen, there could be an option to make the selected item always jump into view (also if `tree-view` is hidden and manually toggled(!)) (in case of feature request)

This behavior is irregardless of wether or not the `tree-view` is visible.

**Comment by Ben3eeE** [comment]
@Redsandro Oh I think I understand now. I was reading the original issue comment like it didn't work when manually revealing by triggering the `tree-view:reveal-active-file` command.

The `autoReveal` option is the option that will automatically scroll the selected file into view. There was an issue that was fixed in Atom 1.27 beta that caused the tree-view to open when `autoReveal` was enabled and switching pane items.

> We like to manually toggle/reveal the tree-view, but (optionally) have the currently selected tab entry centered in the tree-view once we manually toggle the tree-view.

Atom 1.26 with `autoReveal` disabled:
If you want the file to scroll into view -> Manually toggle with `tree-view:reveal-active-file`
If you don't want the file to scroll into view -> Manually toggle with `tree-view:toggle`

With Atom 1.27 the tree-view will no longer open because it's auto revealing with the `autoReveal` setting. So you can leave `autoReveal` enabled and use `tree-view:toggle` to manually toggle when you want.

You can refer to: https://github.com/atom/tree-view/pull/1200 to see how these settings interact with eachother in Atom 1.27.

**Comment by Ben3eeE** [comment]
@Redsandro Does the way these settings interact in Atom 1.27 behave like you expect? You should be able to leave `autoReveal` on and the file be scrolled into view without the tree-view unhiding if you had it hidden.
